### ISSUE TITLE
Generate method should generate inside existing partial `Program` for top-level code

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#60136]
**Version Used**: 

SharpLab  3b32fc3

**Steps to Reproduce**:

```csharp
System.Console.WriteLine();
Program.Test(); // invoke "Generate method 'Test'"

internal partial class Program
{
}
```

**Expected Behavior**:

```csharp
System.Console.WriteLine();
Program.Test();

internal partial class Program
{
    public static void Test()
    {
        throw new System.NotImplementedException();
    }
}
```

**Actual Behavior**:

```csharp
using System;

System.Console.WriteLine();
Program.Test();

void Test()
{
    throw new NotImplementedException();
}

internal partial class Program
{
}
```

---

**Above issue is updated per discussion (https://github.com/dotnet/roslyn/issues/60136#issuecomment-1065902087). The original issue was suggesting to even generate the partial if it doesn't already exist.**

**Comment by CyrusNajmabadi** [comment]
I genuinely don't think this warrants support in Generate-Method.  If anything, we should just block it if it doesn't have a reasonable Type-Declaration to actually generate into.

**Comment by Youssef1313** [comment]
@CyrusNajmabadi At least the case where a `partial Program` is *already* declared should be supported I think?

**Comment by CyrusNajmabadi** [comment]
@Youssef1313 Yup!  Basically, if there is a partial decl, then treat it as a totally normal situation.  If there isn't, treat it as basically not-there, and don't go out of our way to have features generate the partial.

**Comment by Youssef1313** [comment]
@CyrusNajmabadi I updated the issue description and title.
